dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java

```diff
@@ -114,7 +116,18 @@ public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
     }
     // -------------------- PRIVATE --------------------
 
+    /**
+     * Sends an email, Awaitility is necessary since we have to wait until previous transaction will be committed,
+     * otherwise there is a chance that we won't be able to retrieve necessary data from database.
+     */
     private boolean sendEmail(EmailNotificationDto emailNotificationDto, AuthenticatedUser requester) {
```

Comment:
If we would pass only id of notification and do the mapping ` mailMapper.toDto` in this method then it would be more self contained